solenoid_app/views.py

```diff
@@ -151,63 +157,68 @@ def voltageChart(request):
-                                           data['x'], data['force'], data['relative_permeability'], output_unit, data['xGraph'], idv_unit, 0.0, 51.0,
-                                           1.0):
+                                           data['x'], data['force'], data['relative_permeability'], output_unit, data['xGraph'], idv_unit, data['xStart'], data['xEnd'],
+                                           data['xStep']):
                     labels.append(k)
                     graph.append(round(v, sig_figs))
 
             elif data['xGraph'] == 'length':
-                x = 'Length (mm)'
+                x = 'Length'
                 for k, v in solenoid_range(data['voltage'], data['length'], data['r0'], data['ra'], data['awg'],
-                                           data['x'], data['force'], data['relative_permeability'], output_unit, data['xGraph'], idv_unit, 5.0, 26.0,
-                                           1.0):
+                                           data['x'], data['force'], data['relative_permeability'], output_unit, data['xGraph'], idv_unit, data['xStart'], data['xEnd'],
+                                           data['xStep']):
                     labels.append(k)
                     graph.append(round(v, sig_figs))
 
             elif data['xGraph'] == 'r0':
                 x = 'r0'
                 for k, v in solenoid_range(data['voltage'], data['length'], data['r0'], data['ra'], data['awg'],
-                                           data['x'], data['force'], data['relative_permeability'], output_unit, data['xGraph'], idv_unit, 1.0, 5.0,
-                                           0.1):
+                                           data['x'], data['force'], data['relative_permeability'], output_unit, data['xGraph'], idv_unit, data['xStart'], data['xEnd'],
+                                           data['xStep']):
                     labels.append(round(k, sig_figs))
                     graph.append(v)
 
             elif data['xGraph'] == 'ra':
                 x = 'ra'
                 for k, v in solenoid_range(data['voltage'], data['length'], data['r0'], data['ra'], data['awg'],
-                                           data['x'], data['force'], data['relative_permeability'], output_unit, data['xGraph'], idv_unit, 1.0, 5.0,
-                                           0.1):
+                                           data['x'], data['force'], data['relative_permeability'], output_unit, data['xGraph'], idv_unit, data['xStart'], data['xEnd'],
+                                           data['xStep']):
                     labels.append(round(k, sig_figs))
                     graph.append(v)
 
             elif data['xGraph'] == 'x':
+                print(data['xStart'])
                 x = 'x'
-                length = data['length'].magnitude + 1.0
+                if data['length'] is None:
+                    length = 10.0
```

Comment:
Just as a note - in the future we will want this to be just `length = data['length']` when we bugfix the newton's method function in the solver.